models/user.rb

```diff
@@ -24,7 +24,11 @@ def update_settings(params)
     if params['name']
       self.name = params['name']
       raise 'Name cannot be empty' if name.empty?
+    end
 
+    if params['email']
+      self.email = params['email']
+      raise 'Invalid email address' unless /^[^@\s]+@[^@\s]+\.[^@\s]+$/.match?(email)
```

Comment:
didn't you just add this?

Reply:
> didn't you just add this?

It’s the other way round. This PR came first. Later I remembered [your comment](https://github.com/tobymao/18xx/pull/716#issuecomment-644882971) on #716. I’ll merge the changes – presuming you’re fine with the general idea.